javascript-client/package.json

```diff
@@ -26,19 +26,30 @@
     "@typescript-eslint/eslint-plugin": "^4.14.0",
     "@typescript-eslint/parser": "^4.14.0",
     "eslint": "^7.18.0",
+    "eslint-config-airbnb-typescript": "^12.0.0",
+    "eslint-plugin-import": "^2.22.1",
     "npm-run-all": "^4.1.5",
     "rimraf": "^3.0.2",
     "typescript": "^4.1.3"
   },
   "eslintConfig": {
     "parser": "@typescript-eslint/parser",
+    "parserOptions": {
+      "project": "./tsconfig.json"
+    },
     "plugins": [
       "@typescript-eslint"
     ],
     "extends": [
       "eslint:recommended",
       "plugin:@typescript-eslint/eslint-recommended",
-      "plugin:@typescript-eslint/recommended"
-    ]
+      "plugin:@typescript-eslint/recommended",
+      "airbnb-typescript/base"
+    ],
+    "rules": {
+      "@typescript-eslint/no-non-null-assertion": 0,
+      "no-await-in-loop": 0,
```

Comment:
```suggestion
```

I think this is a good rule in general to encourage.

To the extent that we knowingly break it in one place in this code, it's due to very specific rationale around upload bandwidth saturation, not unsoundness in the rule itself. I think it's best to just ignore the single line in this case, rather than globally waive the rule.

Reply:
I've had to disable this enough times now that I've come to just ignore it globally. My attitude is, if I'm using await in a loop, I'm doing it on purpose. I know how to run things in parallel vs. serial, and I think anyone developing on our projects should know that too. This strikes me as a rule to catch more novice developers who don't know about Promise.all.

But I defer to you here. If you find value in this rule rather than just annoyance, let's leave it on.

Reply:
Got it. That seems like a reasonable thing to watch out for as the codebase continues to mature.